packages/govern-console/src/lib/web3-contracts.ts

```diff
@@ -1,52 +1,52 @@
 import { useEffect, useCallback, useMemo, useState, useRef } from 'react'
 import {
+  BigNumber as EthersBigNumber,
   Contract as EthersContract,
+  getDefaultProvider,
   providers,
   utils as EthersUtils,
-  getDefaultProvider,
 } from 'ethers'
-import { useWallet } from 'use-wallet'
+import { useWallet } from '../Providers/Wallet'
 import { getKnownContract } from './known-contracts'
 import { useChainId } from '../Providers/ChainId'
-import { bigNum, getNetworkNode } from './web3-utils'
+import { bigNum, getNetworkNode, getNetworkEthersName } from './web3-utils'
 
 export function useContract(
   address: string,
   abi: any[],
   signer = true,
-): EthersContract {
-  const { ethereum } = useWallet()
+): EthersContract | null {
+  const { wallet } = useWallet()
   const { chainId } = useChainId()
 
-  let ethersProvider
-  if (!ethereum) {
-    ethersProvider = getDefaultProvider(chainId)
-  } else {
-    ethersProvider = new providers.Web3Provider(ethereum)
-  }
+  const ethersProvider = wallet.ethereum
+    ? new providers.Web3Provider(wallet.ethereum)
+    : getDefaultProvider(getNetworkEthersName(chainId))
 
-  if (!address || !EthersUtils.isAddress(address) || !ethersProvider || !abi) {
-    return null
-  }
+  const ethersSignerProvider =
+    signer &&
+    wallet.ethereum &&
+    ethersProvider instanceof providers.Web3Provider
```

Comment:
This is because `getDefaultProvider()` might return `BaseProvider` rather than `Web3Provider`, and it doesn’t have `.getSigner()`.

Reply:
Nice catch!